### ISSUE TITLE
Allow to unset follow mode globally outside helm

### ISSUE THREAD

**Issue by vemv** [emacs-helm/helm#2075]
It appears to be impossible to (cleanly) unset `helm-follow-mode` non-interactively, i.e outside a helm context.

## Rationale

I use https://github.com/syohex/emacs-helm-ag, which offers two kinds of persistent actions, that can be switched between them. If I set follow-mode for one, the other should be unaffected. The simplest way to achieve this would be to clear all 'follow mode' flags before this switching.

Hope it sounds reasonable!

Cheers - Victor

**Comment by thierryvolpiatto** [comment]
vemv <notifications@github.com> writes:

> It appears to be impossible to (cleanly) unset helm-follow-mode
> non-interactively, i.e outside a helm context.

You can do something like this in your init file after loading helm-ag:

    (helm-attrset 'follow nil helm-ag-source)

See also how to modify an attribute in a class (when the source is not
yet defined):

https://github.com/emacs-helm/helm/wiki/FAQ#why-is-a-customizable-helm-source-nil

> Rationale
>
> I use https://github.com/syohex/emacs-helm-ag, which offers two kinds
> of persistent actions, that can be switched between them. If I set
> follow-mode for one, the other should be unaffected. The simplest way
> to achieve this would be to clear all 'follow mode' flags before this
> switching.

You can disable follow-mode for a source but not specifically for an
action.

--
Thierry

**Comment by vemv** [comment]
Hi Thierry, thanks for the support! Much appreciated.

I got the desired behavior almost working, but the one thing missing is: set follow-mode non-interactively, in advance.

i.e. Before invoking `(helm ...)`, letting Helm know that I'll want follow behavior from the very beginning, without need of remembering to hit some shortcut.

Is that possible?

I tried `(helm-attrset 'follow 1 helm-ag-source)`, to no avail.

Thanks - Victor
